added skin prefix

fokkonaut opened the pull request.

fokkonaut committed:
added coala skins (DDCoala)

heinrich5991 commented:
I think the config variable should be a string, e.g. `cl_vanilla_skin_prefix`. It can then be set to an arbitrary value, e.g. "kitty": When the client decides to render a skin, it checks whether the prefix is nonempty, and if it is, it checks for the filename "{prefix}_{skin}" instead of "{skin}" if the prefix isn't already contained in the skin name.

The only other change necessary for this to work is to copy the "brownbear.png" to "coala_brownbear.png".

heinrich5991 commented on src/game/client/components/menus_settings.cpp:
```diff
@@ -424,17 +425,32 @@ void CMenus::RenderSettingsTee(CUIRect MainView)
 	}
+	
+	ReplaceVanillaSkins.HSplitTop(20.0f, &ReplaceVanillaSkinsLabel, &ReplaceVanillaSkins);
+	int ReplaceVanillaSkinsKitty;
```
This should be a static variable, otherwise it could change its address between calls.

heinrich5991 commented on src/game/client/components/menus_settings.cpp:
```diff
@@ -424,17 +425,32 @@ void CMenus::RenderSettingsTee(CUIRect MainView)
+	}
+
+	ReplaceVanillaSkins.HSplitTop(20.0f, &ReplaceVanillaSkinsLabel, &ReplaceVanillaSkins);
+	int ReplaceVanillaSkinsCoala;
```
Should also be static.

fokkonaut commented:
brownbear does not have to be coala_brownbear? there is no such skin because it already looks like a coala (the ears, etc.)

fokkonaut committed:
changed ints to static ints

fokkonaut commented on src/game/client/components/skins.cpp:
```diff
@@ -191,6 +191,20 @@ int CSkins::Find(const char *pName)
+		for(unsigned int i = 0; i < sizeof(s_aaVanillaSkins) / sizeof(s_aaVanillaSkins[0]); i++)
+		{
+			// index 2 and 5 are brownbear and coala
+			if(i != 2 && i != 5 && str_comp_num(pName, s_aaVanillaSkins[i], str_length(s_aaVanillaSkins[i]) - 4) == 0)
```
have a look at this, heinrich. coala and brownbear are excluded

fokkonaut commented:
look at this, heinrich. brownbear and coala are excluded from the transformations
https://github.com/ddnet/ddnet/pull/1218/files#diff-1e43db9e64f6b37386998e364166add8R199

fokkonaut commented:
also i think its better without a string. also the help shows that 1 is kitty and 2 is coala.

heinrich5991 commented:
With a string, the user can add more transformations, without changing the code. I gave an algorithm for how you can decide what skin to pick, but it doesn't work for the brownbear skin, thus I suggested you copy it to "coala_brownbear".

fokkonaut committed:
changed the prefix and added skins

fokkonaut commented:
i added an edit box now. its possible to create own prefix-skinsets now, but there has to be every skin with the prefix_skinname format for it to work. even x_ninja. also i added some more skins (just the rest of kitty and coala and the christmas santa pack so people get an idea of how this works and what it is)

fokkonaut committed:
oop

heinrich5991 committed:
Remove the skin file changes, fix twinbopp -> twinbop

heinrich5991 committed:
Fix duplicate skins not being true duplicates

heinrich5991 committed:
Add fallback for skin prefix, remove duplicate skins

heinrich5991 committed:
Allow changing non-vanilla skins as well
